lib/src/migrators/module.dart

```diff
@@ -125,32 +126,28 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
   /// rule that last imported that URL.
   final _originalImports = <Uri, Tuple2<String, Importer>>{};
 
+  /// Tracks `@import` rules within the entrypoint that could potentially be
+  /// converted to `@forward` instead of `@use`.
+  ///
+  /// If an `@import` rule is visited but none of its members are referenced,
+  /// its canonical URL will be added to this, mapped to the patch that
+  /// converted it to a `@use` rule.
+  ///
+  /// When determining `@forward` rules to add based on --forward, if a URL that
+  /// would have an extra rule is a key in this map, the migrator will instead
+  /// mutate the patch to be a `@forward` rule instead of a `@use` rule.
+  final _possibleForwardConversions = <Uri, Patch>{};
```

Comment:
I don't like storing patches to modify later... I hate to make such a nice self-contained class mutable, and it would be nice to keep the patch list as a write-only data structure as much as possible. To that end, would it be possible to determine in `_migrateImport()` whether a `@use` or a `@forward` should be generated, and just choose the correct one there and then?

Reply:
Yeah, I didn't think it was possible at first, but ultimately realized it was, so existing the new rule for existing `@import`s is now determined in `_migrateImport`.